### PR TITLE
feat(web): refactor sidebar

### PR DESCRIPTION
**Pull request by issmail-basel**
# Overview
Remove duplicated code for header and sidebar

## What I've done
We have two types of sidebars:
- Project sidebar
- Workspace sidebar
You can wrap your page with the same outlines and choose the needed sidebar to show, and which item to toggle.

### CODE REVIEW

**Review by nourbalaha — CHANGES_REQUESTED** [review]

File: web/src/components/organisms/Dashboard/index.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import Header from "@reearth-cms/components/atoms/Header";
 import Layout from "@reearth-cms/components/atoms/Layout";
 import Sider from "@reearth-cms/components/atoms/Sider";
 import MoleculeHeader from "@reearth-cms/components/molecules/Common/Header";
+import ProjectMenu from "@reearth-cms/components/molecules/Common/projectMenu";
```
[the remark this answers is not in the document]
Comment (issmail-basel) [reply web/src/components/organisms/Dashboard/index.tsx]: Done
https://github.com/reearth/reearth-cms/blob/0695816b501e5e92eada82684e7503ba137563c1/web/src/components/organisms/Dashboard/index.tsx#L8-L11

**Review by nourbalaha — APPROVED** [review]

File: web/src/components/organisms/Dashboard/index.tsx [hunk 2]
```diff
@@ -48,11 +50,21 @@ const Dashboard: React.FC<Props> = ({ children, defaultSelectedKeys }) => {
             collapsible
             collapsed={collapsed}
             onCollapse={value => setCollapsed(value)}>
-            <WorkspaceMenu
-              defaultSelectedKeys={defaultSelectedKeys}
-              isPersonalWorkspace={personalWorkspace?.id === currentWorkspace?.id}
-              inlineCollapsed={collapsed}
-              workspaceId={currentWorkspace?.id}></WorkspaceMenu>
+            {projectId && menuType ? (
```
Comment (KaWaite) [inline web/src/components/organisms/Dashboard/index.tsx:53]: Can you think of a better way to conditionally show these two menu types? 

Right now you aren't using the `menuType` in a meaningful way. Even if its `"workspace"`, if there is a `projectId` it'll show the `ProjectMenu~ component, which defeats the purpose of having menuType be `"project" | "workspace"`.

File: web/src/components/organisms/Project/Schema/index.tsx [hunk 3]
```diff
@@ -1,23 +1,15 @@
 import styled from "@emotion/styled";
-import React, { useState } from "react";
+import React from "react";
```
Comment (KaWaite) [inline web/src/components/organisms/Project/Schema/index.tsx:2]: You should be able to remove this completely. I believe CMS is (or should be) setup with the Vite react plugin, which will inject the import React in at build if its not there.

File: web/src/components/organisms/Dashboard/index.tsx (hunk 2 above)
Comment (issmail-basel) [reply web/src/components/organisms/Dashboard/index.tsx:53]: @KaWaite , yup the projectId should wrap the two menus, will fix

File: web/src/components/organisms/Project/Schema/index.tsx [hunk 4]
```diff
@@ -73,56 +53,26 @@ const ProjectSchema: React.FC = () => {
...
-      <WorkspaceCreationModal
-        open={workspaceModalShown}
-        onClose={handleWorkspaceModalClose}
-        onSubmit={handleWorkspaceCreate}
-      />
+        </ContentChild>
+        <FieldListWrapper>
+          <FieldList addField={handleFieldCreationModalOpen} />
+        </FieldListWrapper>
+      </PaddedContent>
```
Comment (KaWaite) [inline web/src/components/organisms/Project/Schema/index.tsx:75]: This file has duplicate FieldUpdateModals. Please delete one

File: web/src/components/organisms/Project/index.tsx [hunk 5]
```diff
@@ -78,6 +78,7 @@ const Project: React.FC = () => {
 const PaddedContent = styled(Content)`
   margin: 16px;
   background-color: #fff;
+  min-height: 100%;
```
Comment (KaWaite) [inline web/src/components/organisms/Project/index.tsx:81]: Do we want the content to be able to be more than 100%? If not, 
```suggestion
  height: 100%;
```

File: web/src/components/organisms/Settings/Members/index.tsx [hunk 6]
```diff
@@ -148,6 +148,7 @@ const Members: React.FC = () => {
 const PaddedContent = styled(Content)`
   margin: 16px;
   background-color: #fff;
+  min-height: 100%;
```
Comment (KaWaite) [inline web/src/components/organisms/Settings/Members/index.tsx:151]: Same here. If min-height is needed, you can disregard this comment.

File: web/src/components/organisms/Dashboard/index.tsx (hunk 2 above)
Comment (KaWaite) [reply web/src/components/organisms/Dashboard/index.tsx:53]: One option is pass the menu component in as a prop.

File: web/src/components/organisms/Settings/Members/index.tsx (hunk 6 above)
Comment (issmail-basel) [reply web/src/components/organisms/Settings/Members/index.tsx:151]: @KaWaite , yes it is
Figma requires to have the center content to be white and full height in certain pages

File: web/src/components/organisms/Dashboard/index.tsx (hunk 2 above)
Comment (issmail-basel) [reply web/src/components/organisms/Dashboard/index.tsx]: @KaWaite , I thought of passing it, but the whole project has 2 menus, and giving it as a prop will add an unnecessary import and prop passing for each menu usage.

https://github.com/reearth/reearth-cms/blob/6e45c9fb406680bf370be1b073859163593f42ba/web/src/components/organisms/Dashboard/index.tsx#L49-L73

File: web/src/components/organisms/Project/Schema/index.tsx (hunk 4 above)
Comment (issmail-basel) [reply web/src/components/organisms/Project/Schema/index.tsx:76]: @KaWaite , I couldn't find the duplication, please can you explain more?

File: web/src/components/organisms/Project/Schema/index.tsx (hunk 3 above)
Comment (issmail-basel) [reply web/src/components/organisms/Project/Schema/index.tsx:2]: https://github.com/reearth/reearth-cms/blob/6e45c9fb406680bf370be1b073859163593f42ba/web/src/components/organisms/Dashboard/index.tsx#L2

File: web/src/components/organisms/Settings/Members/index.tsx (hunk 6 above)
Comment (issmail-basel) [reply web/src/components/organisms/Settings/Members/index.tsx]: done.

File: web/src/components/organisms/Project/index.tsx (hunk 5 above)
Comment (issmail-basel) [reply web/src/components/organisms/Project/index.tsx]: done.

File: web/src/components/organisms/Dashboard/index.tsx [hunk 7]
```diff
@@ -48,11 +50,26 @@ const Dashboard: React.FC<Props> = ({ children, defaultSelectedKeys }) => {
...
+                )}
+              </>
+            ) : (
+              <WorkspaceMenu
+                defaultSelectedKeys={defaultSelectedKeys}
+                isPersonalWorkspace={personalWorkspace?.id === currentWorkspace?.id}
+                inlineCollapsed={collapsed}
+                workspaceId={currentWorkspace?.id}
+              />
+            )}
```
Comment (KaWaite) [inline web/src/components/organisms/Dashboard/index.tsx:72]:
```suggestion
            {menuType === "project" ?
                projectId && (
                  <ProjectMenu
                    projectId={projectId}
                    defaultSelectedKeys={defaultSelectedKeys}
                    inlineCollapsed={collapsed}
                    workspaceId={currentWorkspace?.id}
                  />
                )
              </>
            ) : (
              <WorkspaceMenu
                defaultSelectedKeys={defaultSelectedKeys}
                isPersonalWorkspace={personalWorkspace?.id === currentWorkspace?.id}
                inlineCollapsed={collapsed}
                workspaceId={currentWorkspace?.id}
              />
            )}
```

Comment (issmail-basel) [reply web/src/components/organisms/Dashboard/index.tsx:72]: @KaWaite , you can't nest conditions like that, adding a fragment is a must here.

File: web/src/components/organisms/Project/Schema/index.tsx (hunk 4 above)
Comment (KaWaite) [reply web/src/components/organisms/Project/Schema/index.tsx:76]: Line 134-151

Comment (issmail-basel) [reply web/src/components/organisms/Project/Schema/index.tsx:76]: @KaWaite , file is 139 lines long, I think you are viewing the deleted lines?

File: web/src/components/organisms/Dashboard/index.tsx (hunk 7 above)
Comment (issmail-basel) [reply web/src/components/organisms/Dashboard/index.tsx]: https://github.com/reearth/reearth-cms/blob/339fdc4cd4270b04c91816c97546e47b97c1a211/web/src/components/organisms/Dashboard/index.tsx#L53-L69

File: web/src/components/organisms/Settings/Members/index.tsx (hunk 6 above)
Comment (issmail-basel) [reply web/src/components/organisms/Settings/Members/index.tsx]: actually, min-height is required, I reverted to it after testing.
yes, content can be more than 100%.

**Review by KaWaite — APPROVED** [review]

**Pull request merged**